File: Chapter 05/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter05/section_5_4_server_monitor/after/tests/ServerMonitorTests.java
package com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.after.tests;

import com.grokkingsoftwarearchitecture.chapter05.section_5_4_server_monitor.after.core.domain.ServerMonitor;

/**
 * ARCHITECTURAL TEST
 * Proves that by using Ports, we have restored our Testability.
 */
public class ServerMonitorTests {

    private ServerMonitorTests() {
        // Private constructor to hide the implicit public one
    }

    public static void run() {
        System.out.println("--- RUNNING ARCHITECTURAL TEST: HEXAGONAL ---");
        testServerOverheatingSendsAlertExactlyOnce();
    }

    private static void testServerOverheatingSendsAlertExactlyOnce() {
        // 1. Arrange
        // We use the FakeAlertPort instead of a real API.
        FakeAlertPort fakePort = new FakeAlertPort();
        ServerMonitor monitor = new ServerMonitor(fakePort);

        // 2. Act
        System.out.println("Test Action: Checking temperature at 96 degrees...");
        monitor.checkTemperature(96);

        // 3. Assert
        if (fakePort.getSentMessages().size() == 1 && 
            fakePort.getSentMessages().get(0).contains("Take cover")) {
            System.out.println("SUCCESS: Alert sent correctly to the Port.");
        } else {
            System.out.println("FAIL: Alert logic failed verification.");
        }
    }
}

File: Chapter 05/C#/Section_5_4_Server_Monitor/After/Tests/ServerMonitorTests.cs
using System;
using Chapter05.ServerMonitor.After.Core.Domain;

// ALIAS FIX: Tells the compiler exactly which ServerMonitor to use
using MonitorClass = Chapter05.ServerMonitor.After.Core.Domain.ServerMonitor;

namespace Chapter05.ServerMonitor.After.Tests
{
    /// <summary>
    /// ARCHITECTURAL TEST
    /// Fulfills the Scribe role by proving the test passes without hitting a real API.
    /// </summary>
    public static class ServerMonitorTests
    {
        public static void Run()
        {
            Console.WriteLine("--- RUNNING ARCHITECTURAL TEST: HEXAGONAL ---");
            
            // Arrange
            var fakePort = new FakeAlertPort();
            
            // Using the alias to avoid the namespace collision
            var monitor = new MonitorClass(fakePort);

            // Act
            Console.WriteLine("Test Action: Checking temperature at 96 degrees...");
            monitor.CheckTemperature(96);

            // Assert
            if (fakePort.SentMessages.Count == 1 && fakePort.SentMessages[0].Contains("Take cover"))
            {
                Console.WriteLine("SUCCESS: Alert sent correctly to the Port.");
            }
            else
            {
                Console.WriteLine("FAIL: Alert logic failed verification.");
            }
        }
    }
}